cli/src/lockfiles/mod.rs

```diff
@@ -20,12 +19,15 @@ pub use ruby::GemLock;
 
 pub type ParseResult = anyhow::Result<Vec<PackageDescriptor>>;
 
-pub trait Parseable {
-    fn new(filename: &Path) -> Result<Self, io::Error>
-    where
-        Self: Sized;
-
-    fn parse(&self) -> ParseResult;
+pub trait Parse {
+    /// Parse from a string
+    fn parse(&self, data: &str) -> ParseResult;
+    /// Indicate the type of packages parsed by this parser
+    fn package_type(&self) -> PackageType;
```

Comment:
Could this still just be a static function? Not a big deal either way, but I think it makes more sense semantically to not have the reference to self. This function only cares about the implementing type, not the instance.

Reply:
Non-methods cannot be dispatched from trait objects (see [object safety](https://doc.rust-lang.org/reference/items/traits.html#object-safety)). So if this was a static function and we have something of type `&dyn Parse`, we would have no way to call this function.

Reply:
Right, my mistake